src/test/scala/edu/ie3/osmogrid/lv/coordinator/LvGraphBuilderSpec.scala

```diff
@@ -0,0 +1,242 @@
+        val expectedOrthogonal = GeoUtils.buildCoordinate(50d, 7.5)
+        val minDistance = 0.05.asKilometre
+        val getClosest =
+          PrivateMethod[Try[(ComparableQuantity[Length], Node)]](
+            Symbol("getClosest")
+          )
+        LvGraphBuilder invokePrivate getClosest(
+          lineNodeA,
+          lineNodeB,
+          buildingCenter,
+          minDistance
+        ) match {
+          case Success(distanceAndNode) =>
+            distanceAndNode._1 should equalWithTolerance(
+              buildingCenter.haversineDistance(expectedOrthogonal)
+            )
+            distanceAndNode._2.latitude shouldBe (expectedOrthogonal.y +- 1e-8)
+            distanceAndNode._2.longitude shouldBe (expectedOrthogonal.x +- 1e-8)
+          case Failure(exc) => fail(s"Test failed due to exception: ", exc)
+        }
+      }
+
+      "calculate an orthogonal projection correctly" in {
+        val coordinateA = GeoUtils.buildCoordinate(50d, 7)
+        val coordinateB = GeoUtils.buildCoordinate(50d, 8)
+        val point = GeoUtils.buildCoordinate(49d, 7.5)
+        val actual = orthogonalProjection(
+          coordinateA,
+          coordinateB,
+          point
+        )
+        actual.x shouldBe (7.5 +- 1e-9)
+        actual.y shouldBe (50d +- 1e-9)
+      }
+
+      "update the street graph with all building graph connections correctly" when {
+        val connectingNode = Node(
+          99L,
+          (nodes.highway1Node1.latitude + nodes.highway1Node2.latitude) / 2,
+          (nodes.highway1Node1.longitude + nodes.highway1Node2.longitude) / 2,
+          Map.empty,
+          None
+        )
+        val buildingGraphConnection = BuildingGraphConnection(
+          ways.building1,
+          mock[Coordinate],
+          10.asKiloWatt,
+          nodes.highway1Node1,
+          nodes.highway1Node2,
+          connectingNode
+        )
+        val updateGraphWithBuildingConnections =
+          PrivateMethod[OsmGraph](Symbol("updateGraphWithBuildingConnections"))
+
+        "not considering building graph connections these are not added to the graph" in {
+          val osmGraph = new OsmGraph()
+          osmGraph.addVertex(nodes.highway1Node1)
+          osmGraph.addVertex(nodes.highway1Node2)
+          osmGraph.addWeightedEdge(nodes.highway1Node1, nodes.highway1Node2)
+          val actual: OsmGraph =
+            LvGraphBuilder invokePrivate updateGraphWithBuildingConnections(
+              osmGraph,
+              Seq(buildingGraphConnection).par,
+              true
+            )
+          actual.vertexSet().size() shouldBe 4
+          actual.containsVertex(connectingNode) shouldBe true
+          actual.containsEdge(
+            nodes.highway1Node1,
+            nodes.highway1Node2
+          ) shouldBe false
+          actual.containsEdge(nodes.highway1Node1, connectingNode) shouldBe true
+          actual.containsEdge(connectingNode, nodes.highway1Node2) shouldBe true
+        }
```

Comment:
What about checking if graph contains an edge between connectingNode and buildingCenter and calculating its length?